src/time_series_manager.jl

```diff
@@ -0,0 +1,275 @@
+        LOG_GROUP_TIME_SERIES
+    return
+end
+
+function clear_time_series!(mgr::TimeSeriesManager)
+    _throw_if_read_only(mgr)
+    clear_metadata!(mgr.metadata_store)
+    clear_time_series!(mgr.data_store)
+end
+
+function clear_time_series!(mgr::TimeSeriesManager, component::TimeSeriesOwners)
+    _throw_if_read_only(mgr)
+    for metadata in list_metadata(mgr.metadata_store, component)
+        remove_time_series!(mgr, component, metadata)
+    end
+    @debug "Cleared time_series in $(summary(component))." _group =
+        LOG_GROUP_TIME_SERIES
+    return
+end
+
+has_time_series(mgr::TimeSeriesManager, component::TimeSeriesOwners) =
+    has_time_series(mgr.metadata_store, component)
+has_time_series(::Nothing, component::TimeSeriesOwners) = false
+
+function Base.deepcopy_internal(mgr::TimeSeriesManager, dict::IdDict)
+    if haskey(dict, mgr)
+        return dict[mgr]
+    end
+    data_store = deepcopy(mgr.data_store)
+    if mgr.data_store isa Hdf5TimeSeriesStorage
+        copy_to_new_file!(data_store, dirname(mgr.data_store.file_path))
+    end
+
+    new_db_file = backup_to_temp(mgr.metadata_store)
+    metadata_store = TimeSeriesMetadataStore(new_db_file)
+    new_mgr = TimeSeriesManager(data_store, metadata_store, mgr.read_only)
+    dict[mgr] = new_mgr
+    dict[mgr.data_store] = new_mgr.data_store
+    dict[mgr.metadata_store] = new_mgr.metadata_store
+    return new_mgr
+end
+
+get_metadata(
+    mgr::TimeSeriesManager,
+    component::TimeSeriesOwners,
+    time_series_type::Type{<:TimeSeriesData},
+    name::String;
+    features...,
+) = get_metadata(
+    mgr.metadata_store,
+    component,
+    time_series_type,
+    name;
+    features...,
+)
+
+list_metadata(
+    mgr::TimeSeriesManager,
+    component::TimeSeriesOwners;
+    time_series_type::Union{Type{<:TimeSeriesData}, Nothing} = nothing,
+    name::Union{String, Nothing} = nothing,
+    features...,
+) = list_metadata(
+    mgr.metadata_store,
+    component;
+    time_series_type = time_series_type,
+    name = name,
+    features...,
+)
+
+"""
+Remove the time series data for a component.
+"""
+function remove_time_series!(
+    mgr::TimeSeriesManager,
+    time_series_type::Type{<:TimeSeriesData},
+    component::TimeSeriesOwners,
+    name::String;
+    features...,
+)
+    _throw_if_read_only(mgr)
+    uuids = list_matching_time_series_uuids(
+        mgr.metadata_store;
+        time_series_type = time_series_type,
+        name = name,
+        features...,
+    )
+    remove_metadata!(
+        mgr.metadata_store,
+        component;
+        time_series_type = time_series_type,
+        name = name,
+        features...,
+    )
+
+    @debug "Removed time_series metadata in $(summary(component))." _group =
+        LOG_GROUP_TIME_SERIES component time_series_type name features
+
+    for uuid in uuids
+        _remove_data_if_no_more_references(mgr, uuid)
+    end
+
+    return
+end
+
+function remove_time_series!(
```

Comment:
Is this anything?
<img width="1071" alt="Screenshot 2024-04-25 at 5 01 18 PM" src="https://github.com/NREL-Sienna/InfrastructureSystems.jl/assets/23368820/545926f0-011b-455b-b6d8-2feec54af390">
Setup along the lines of `@testset "Test add SingleTimeSeries with features"` up to and including the first `IS.add_time_series!`

Reply:
This is now fixed. Thanks.